src/globus_sdk/experimental/globus_app/validating_token_storage/context.py

```diff
@@ -0,0 +1,16 @@
+from __future__ import annotations
+
+import dataclasses
+
+
+@dataclasses.dataclass
+class TokenValidationContext:
+    """
+    This is the context object which will be passed to token validators.
+
+    It allows the ValidatingTokenStorage to capture state about its execution and pass
+    that information, nicely bundled, to validators.
+    """
+
+    prior_identity_id: str | None
+    token_data_identity_id: str | None
```

Comment:
It is not, but given that there's a minor perf penalty for using frozen dataclasses, I thought it would be fine to define it as such. (Since most objects are mutable) 

The perf hit is very minor though. I can mark it frozen if you like?

Reply:
I consider all dataclasses to be a performance penalty, so my preference would be to make this a named tuple.

```python
class TokenValidationContext(typing.NamedTuple):
    """
    This is the context object which will be passed to token validators.
    It allows the ValidatingTokenStorage to capture state about its execution and pass
    that information, nicely bundled, to validators.
    """

    prior_identity_id: str | None
    token_data_identity_id: str | None
```

Reply:
I need to think about this... To put my thoughts into text:

We currently have several datatype construction paradigms in the SDK (by my count, 3: dataclass, PayloadWrapper, and Serializable).
One is "our old customized UserDict", one was "built for GAREs but turned out to have other valid use-cases", and one is "'the' stdlib construct we use". Even if it's superior in this case, introducing namedtuples adds a fourth.
Each was well-justified in its own context, and if you need 4 things to express what you mean, then you need 4 things. Arguably there's no issue.

I think the only thing I dislike is that we would have namedtuples _and_ frozen dataclasses. If at all possible, we should choose between those two. (And I'm onboard with choosing namedtuple.)

Reply:
The only frozen dataclass currently in use is the `GlobusAppConfig`.

I'll raise this with the team after my some meetings; I'm not sure if there would be any harm in making that also a typing.NamedTuple.

Reply:
GlobusAppConfig uses a `defaultfactory` to set a default environment ([ref](https://github.com/globus/globus-sdk-python/blob/main/src/globus_sdk/experimental/globus_app/config.py#L93)); I don't believe there's any equivalent functionality in NamedTuple.

Reply:
There is not. The only way to get that with a namedtuple, as far as I know, is to have a builder of some kind which makes the whole object.

I'll bring this up in the team chat -- the question I have is whether or not "minor perf hit but fewer different types of objects" is a good reason to make this object a dataclass (frozen or otherwise) rather than a named tuple. Obviously either one "works"; the open question is which one is _best_.